exporter/googlecloudexporter/config.go

```diff
@@ -0,0 +1,175 @@
+	return &Config{
+		ExporterSettings: config.NewExporterSettings(config.NewComponentID(typeStr)),
+		Location:         defaultLocation,
+		Namespace:        defaultNamespace,
+		GCPConfig:        createDefaultGCPConfig(),
+		BatchConfig:      createDefaultBatchConfig(),
+		NormalizeConfig:  createDefaultNormalizerConfig(),
+		DetectorConfig:   createDefaultDetectorConfig(),
+		AttributerConfig: createDefaultAttributerConfig(),
+		TransposerConfig: createDefaultTransposerConfig(),
+	}
+}
+
+// createDefaultGCPConfig creates a default GCP config
+func createDefaultGCPConfig() *gcp.Config {
+	gcpFactory := gcp.NewFactory()
+	gcpConfig := gcpFactory.CreateDefaultConfig().(*gcp.Config)
+	gcpConfig.RetrySettings.Enabled = false
+	gcpConfig.UserAgent = defaultUserAgent
+	gcpConfig.MetricConfig.Prefix = defaultMetricPrefix
+	gcpConfig.ResourceMappings = []gcp.ResourceMapping{
+		{
+			TargetType: genericNodeResource,
+			LabelMappings: []gcp.LabelMapping{
+				{
+					SourceKey: "host.name",
+					TargetKey: "node_id",
+				},
+				{
+					SourceKey: "location",
+					TargetKey: "location",
+				},
+				{
+					SourceKey: "namespace",
+					TargetKey: "namespace",
+				},
+			},
+		},
+	}
+
+	return gcpConfig
+}
+
+// createDefaultBatchConfig creates a default batch config
+func createDefaultBatchConfig() *batchprocessor.Config {
+	batchFactory := batchprocessor.NewFactory()
+	batchConfig := batchFactory.CreateDefaultConfig().(*batchprocessor.Config)
+	return batchConfig
+}
+
+// createDefaultNormalizerConfig creates a default normalizer config
+func createDefaultNormalizerConfig() *normalizesumsprocessor.Config {
+	normalizeFactory := normalizesumsprocessor.NewFactory()
+	return normalizeFactory.CreateDefaultConfig().(*normalizesumsprocessor.Config)
+}
+
+// createDefaultDetectorConfig creates a default detector config
+func createDefaultDetectorConfig() *resourcedetectionprocessor.Config {
+	detectorFactory := resourcedetectionprocessor.NewFactory()
+	detectorConfig := detectorFactory.CreateDefaultConfig().(*resourcedetectionprocessor.Config)
+	detectorConfig.Detectors = []string{"system"}
+	detectorConfig.DetectorConfig.SystemConfig.HostnameSources = []string{"os"}
+	return detectorConfig
+}
+
+// createDefaultAttributerConfig creates a default attributer config
+func createDefaultAttributerConfig() *resourceprocessor.Config {
+	attributerFactory := resourceprocessor.NewFactory()
+	attributerConfig := attributerFactory.CreateDefaultConfig().(*resourceprocessor.Config)
+	return attributerConfig
+}
+
+// createDefaultTransposerConfig creates a default transposer config
+func createDefaultTransposerConfig() *resourceattributetransposerprocessor.Config {
+	transposerFactory := resourceattributetransposerprocessor.NewFactory()
+	transposerConfig := transposerFactory.CreateDefaultConfig().(*resourceattributetransposerprocessor.Config)
+	transposerConfig.Operations = []resourceattributetransposerprocessor.CopyResourceConfig{
+		{
```

Comment:
I'm not as knowledgeable on these but are these relative to the host receiver? If not where do they come from?

Reply:
Yes, they're relative to the host receiver. @jsirianni compiled this initial list. I imagine this mapping will grow with subsequent sources. But for now, I wanted to keep it simple. Especially if the Google Cloud exporter is updated to handle resource attributes better. That will allow us to remove this processor altogether.